pkg/apis/autoscaling/annotation_validation.go

```diff
@@ -43,21 +41,47 @@ func ValidateAnnotations(annotations map[string]string) *apis.FieldError {
 		return nil
 	}
 
+	return validateMinMaxScale(annotations).Also(validateWindows(annotations))
+}
+
+func validateWindows(annotations map[string]string) *apis.FieldError {
+	var errs *apis.FieldError
+	if v, ok := annotations[PanicWindowPercentageAnnotationKey]; ok {
+		if fv, err := strconv.ParseFloat(v, 64); err != nil {
+			errs = errs.Also(apis.ErrInvalidValue(v, PanicWindowPercentageAnnotationKey))
+		} else if fv < PanicWindowPercentageMin || fv > PanicWindowPercentageMax {
+			errs = apis.ErrOutOfBoundsValue(v, PanicWindowPercentageMin,
+				PanicWindowPercentageMax, PanicWindowPercentageAnnotationKey)
+		}
+	}
+	if v, ok := annotations[PanicThresholdPercentageAnnotationKey]; ok {
+		if fv, err := strconv.ParseFloat(v, 64); err != nil {
+			errs = errs.Also(apis.ErrInvalidValue(v, PanicThresholdPercentageAnnotationKey))
+		} else if fv < PanicThresholdPercentageMin || fv > PanicThresholdPercentageMax {
+			errs = errs.Also(apis.ErrOutOfBoundsValue(v, PanicThresholdPercentageMin, PanicThresholdPercentageMax,
+				PanicThresholdPercentageAnnotationKey))
+		}
+	}
+	return errs
+}
+
+func validateMinMaxScale(annotations map[string]string) *apis.FieldError {
+	var errs *apis.FieldError
+
 	min, err := getIntGE0(annotations, MinScaleAnnotationKey)
 	if err != nil {
-		return err
+		errs = err
```

Comment:
Does `errs.Also()` require that `errs` is non-`nil`? If not, you can skip the `if err != nil` block, and just do:

```golang
min, err := getIntGE0(annotations, MinScaleAnnotationKey)
errs = errs.Also(err)

max, err := ...
errs = errs.Also(err)
```

Reply:
Yes, `errs.Also()` can work with nil.
Done.